siobrultech_protocols/gem/protocol.py

```diff
@@ -47,17 +47,18 @@ def __init__(
         self._transport: Optional[asyncio.BaseTransport] = None
 
     def connection_made(self, transport: asyncio.BaseTransport) -> None:
+        LOG.info("Connection opened")
         self._transport = transport
 
-    def connection_lost(self, exc: Optional[Any]) -> None:
+    def connection_lost(self, exc: Optional[BaseException]) -> None:
         if exc is not None:
-            LOG.warning("Connection lost: {}".format(exc))
+            LOG.warning("Connection lost due to exception", exc)
         else:
             LOG.info("Connection closed")
         self._transport = None
 
     def data_received(self, data: bytes) -> None:
-        LOG.debug("Received {} bytes".format(len(data)))
+        LOG.debug(f"Received {len(data)} bytes")
```

Comment:
nit: Old-style formatting is preferable for `logging`, so that it doesn't happen when logging is off or set to a higher level.